rollup/fees/rollup_fee.go

```diff
@@ -203,11 +205,22 @@ func calculateEncodedRollupFeeFeeFeynman(
 	execScalar *big.Int,
 	blobScalar *big.Int,
 ) *big.Int {
-	// tx size (RLP-encoded)
 	txSize := big.NewInt(int64(len(data)))
 
-	// compression_ratio(tx) = 1 (placeholder, scaled to match scalars precision)
-	compressionRatio := big.NewInt(rcfg.Precision.Int64())
+	// Default compression ratio is 1.0 (no compression)
+	compressionRatioInt := big.NewInt(rcfg.Precision.Int64())
+
+	if len(data) != 0 {
+		compressedBytes, err := zstd.CompressScrollBatchBytes(data)
+		if err != nil {
+			log.Error("Compress batch compress failed, using 1.0", "err", err)
+		} else {
+			compressedSize := big.NewInt(int64(len(compressedBytes)))
+			// compressionRatioInt = (compressedSize * precision) / txSize
+			compressionRatioInt.Mul(compressedSize, rcfg.Precision)
+			compressionRatioInt.Div(compressionRatioInt, txSize)
```

Comment:
While I believe it's always true that `compressedSize <= size`, I'm not 100% convinced this is the case. I suggest you add a check to cover this